cpp/inline/mcts/SearchThread.inl

```diff
@@ -622,6 +638,18 @@ int SearchThread<Game>::get_best_child_index(Node* node) {
   return argmax_index;
 }
 
+template <core::concepts::Game Game>
+int SearchThread<Game>::sample_chance_child_index(Node* node) {
+  bool is_root = (node == shared_data_->get_root_node());
+  const SearchParams& search_params = shared_data_->search_params;
+  ActionSelector action_selector(*manager_params_, search_params, node, is_root);
+  const auto* chance_dist = action_selector.chance_dist;
+
+  int n = Game::Constants::kMaxBranchingFactor;
+  core::action_t chance_action = util::Random::weighted_sample(chance_dist, chance_dist + n);
+  return chance_action;
```

Comment:
It makes no sense to construct an `ActionSelector` here.

Also, your choice of `n` is not right - it might be too big for certain games.

You can just replace this entire function implementation with:

```
int n = node->stable_date().num_valid_actions;
float chance_dist[n];
for (int i = 0; i < n; ++i) {
  chance_dist[n] = node->get_edge(i)->chance_prob;
}
return util::Random::weighted_sample(chance_dist, chance_dist + n);
```